subiquity/server/snapd/types.py

```diff
@@ -351,11 +351,13 @@ class InsufficientEntropyReasons(enum.Enum):
 class InsufficientEntropyDetails:
     reasons: List[InsufficientEntropyReasons]
     entropy_bits: float
-    min_entropy_bits: float
+    min_entropy_bits: int
```

Comment:
question(non-blocking): Ideally the mixed type situation will resolve early enough that we'll only have to support the int/int case but if not, can we do:
```python
entropy_bits: int | float
min_entropy_bits: int
```

and have the serializer just do the right thing?

Definitely not something we have to deal with right now, but I was trying to anticipate what the future changes might look like and I haven't worked with the serializer code enough to know how much of a pain it's going to be, if at all.

Reply:
Yes I agree this would be an improvement. I've been willing to make similar changes to the (de)serializer to support more union types but have run into obstacles when I tried. Maybe it would be easier if we start only with the deserialization implementation (this is what we'd need here).

Reply:
It feels like union support for scalars out to be reasonably simple? But I also feel like unions in your API are a bad smell so...

Reply:
I'll try to get this implemented in a separate PR!